avocado/utils/linux_modules.py

```diff
@@ -208,34 +208,91 @@ def get_loaded_modules():
         return [astring.to_text(_.split(b' ', 1)[0]) for _ in proc_modules]
 
 
-def check_kernel_config(config_name):
+def check_kernel_config(config_name, session=None):
     """
     Reports the configuration of $config_name of the current kernel
 
     :param config_name: Name of kernel config to search
+    :param session: guest session, command is run on host if None
     :type config_name: str
     :return: Config status in running kernel (NOT_SET, BUILTIN, MODULE)
     :rtype: :class:`ModuleConfig`
     """
+    if session is None:
+        kernel_version = platform.uname()[2]
+        config_file = '/boot/config-' + kernel_version
+        # Sometimes a kernel config's name may contain another kernel config
+        # E.g.
+        # CONFIG_PREEMPT_VOLUNTARY=y
+        # CONFIG_PREEMPT is not set
+        # So we need to search CONFIG_XXX= in kconfig file
+        config_info = process.run('grep {}= {}'.format(config_name,
```

Comment:
Running a command on the local system to filter a text file should only be used as a last resort (if the text file is in such as format that is not feasible to be parsed easily).  Of course, it's a different story when operating on a session to a remote system (there's no other way around it).

I suggest refactoring the previously existing code into its own helper method and reusing it here.

Reply:
> Running a command on the local system to filter a text file should only be used as a last resort (if the text file is in such as format that is not feasible to be parsed easily). Of course, it's a different story when operating on a session to a remote system (there's no other way around it).
> 
> I suggest refactoring the previously existing code into its own helper method and reusing it here.

Ok, I got it. My original idea was to have the host and session use the same method to get kernel config info.
Then, we could use common codes to determine which state the kernel config is. (NOT_SET, BUILTIN, MODULE) 

Now, I think your suggestion is much better.
I will add 2 methods: check_{host, guest}_kernel_config. The check_host_kernel_config will use the
previously existing code.